lib/rabbitmq/cli/ctl/commands/ping_command.ex

```diff
@@ -0,0 +1,77 @@
+## The contents of this file are subject to the Mozilla Public License
+## Version 1.1 (the "License"); you may not use this file except in
+## compliance with the License. You may obtain a copy of the License
+## at http://www.mozilla.org/MPL/
+##
+## Software distributed under the License is distributed on an "AS IS"
+## basis, WITHOUT WARRANTY OF ANY KIND, either express or implied. See
+## the License for the specific language governing rights and
+## limitations under the License.
+##
+## The Original Code is RabbitMQ.
+##
+## The Initial Developer of the Original Code is Pivotal Software, Inc.
+## Copyright (c) 2016-2018 Pivotal Software, Inc.  All rights reserved.
+
+
+defmodule RabbitMQ.CLI.Ctl.Commands.PingCommand do
+  @behaviour RabbitMQ.CLI.CommandBehaviour
+
+  @default_timeout 60_000
+
+  def merge_defaults(args, opts) do
+    timeout = case opts[:timeout] do
+      nil       -> @default_timeout;
+      :infinity -> @default_timeout;
+      other     -> other
+    end
+    {args, Map.merge(opts, %{timeout: timeout})}
+  end
+
+  def switches(), do: [timeout: :integer]
+  def aliases(), do: [t: :timeout]
+
+  def validate([_], _), do: {:validation_failure, :too_many_args}
+  def validate([], _), do: :ok
+
+  def run([], %{node: node_name, timeout: timeout}) do
+    # this is very similar to what net_adm:ping/1 does reimplemented with support for custom timeouts
+    # and error values that are used by CLI commands
+    msg = "Failed to successfully connect and authenticate to #{node_name} in #{timeout} ms"
+    try do
+      case :gen.call({:net_kernel, node_name}, :'$gen_call', {:is_auth, node_name}, timeout) do
```

Comment:
`net_adm:ping` uses the current node name as the argument to `is_auth` ([source](https://github.com/erlang/otp/blob/1526eaead833b3bdcd3555a12e2af62c359e7868/lib/kernel/src/net_adm.erl#L64)) ... do we want to do the same?

Reply:
Curiously [the argument seems to be ignored](https://github.com/erlang/otp/blob/46a07a914f049748e03ed78769ba4310753322e0/lib/kernel/src/net_kernel.erl#L540) but sure, it makes certain sense to me even then.

Reply:
Good one, I didn't think to check the "other side"